src/Stryker.Core/Stryker.Core/Mutants/MutantOrchestrator.cs

```diff
@@ -76,72 +216,66 @@ public SyntaxNode Mutate(SyntaxNode currentNode)
                 foreach (var expressionSyntax in expressions)
                 {
                     var currentExpressionSyntax = currentNodeCopy.GetCurrentNode(expressionSyntax);
-                    if (currentNode is ExpressionStatementSyntax expressionStatement)
+                    SyntaxNode actualMutationNode = null;
+                    if (currentNode is AssignmentExpressionSyntax assign)
                     {
-                        if (GetExpressionSyntax(expressionStatement) is var subExpressionSyntax && subExpressionSyntax != null)
-                        {
-                            // The expression of a ExpressionStatement cannot be mutated directly
-                            return currentNodeCopy.ReplaceNode(currentExpressionSyntax, Mutate(currentExpressionSyntax));
-                        }
-                        else
-                        {
-                            // If the EpxressionStatement does not contain a expression that can be mutated with conditional expression...
-                            // it should be mutated with if statements
-                            return MutateWithIfStatements(currentNode as ExpressionStatementSyntax);
-                        }
+                        actualMutationNode = assign.Right;
+                    }
+                    else if (currentNode is ExpressionStatementSyntax)
```

Comment:
This should be placed outside of the foreach. It checks the same node on every loop. It could be placed at the start of the method?

Reply:
well.... I left the code as I found it, and I must say it did hurt my eyes too. Actually, there was several other constructions that were similar and I was able to fix them.
This one is harder than it looks to get right, and I had to stop refactoring the code at some stage.

Reply:
working on this

Reply:
got a nicer version. It needs further redesign, logic still feels a bit strange, but it is far much better than it originally was